File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0, ans = 0;
        for (int i = 1; i  < prices.length; ++i){
            if (prices[i] < prices[left]){ 
                left = i;
            }else{
                ans = Math.max(ans, prices[i] - prices[left]);
            }
        }
        return ans;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        max_sofar = prices[n- 1]
        ans = 0

        for i in range(n - 2,  -1, -1):
            max_sofar = max(max_sofar , prices[i])
            ans = max(ans, max_sofar - prices[i])
            
        return ans
            
        